enumerables.rb

```diff
@@ -0,0 +1,186 @@
+      end
+      true
+    elsif default.class == Class
+      elements.my_each do |i|
+        return false unless i.class == default
+      end
+      true
+    elsif default.class == Regexp
+      elements.my_each do |i|
+        return false unless i =~ default
+      end
+      true
+    elsif block_given?
+      elements.my_each do |i|
+        return false unless yield i
+      end
+      true
+    elsif !block_given? && !default
+      elements.my_each do |i|
+        return false unless i
+      end
+      true
+    elsif default.class != Regexp && default.class != Class
+      elements.my_each do |i|
+        return false unless i == default
+      end
+      true
+    end
+  end
+
+  def my_any?(default = nil)
+    elements = to_a
+    if default == Numeric
+      elements.my_each do |i|
+        return true if i.class == Integer || i.class == Float || i.Complex
+      end
+      false
+    elsif default.class == Class
+      elements.my_each do |i|
+        return true if i.class == default
+      end
+      false
+    elsif default.class == Regexp
+      elements.my_each do |i|
+        return true if i =~ default
+      end
+      false
+    elsif block_given?
+      elements.my_each do |i|
+        return true if yield i
+      end
+      false
+    elsif !block_given? && !default
+      elements.my_each do |i|
+        return true if i
+      end
+      false
+    elsif default.class != Regexp && default.class != Class
+      elements.my_each do |i|
+        return true if i == default
+      end
+      false
+    end
+  end
+
+  def my_none?(default = nil)
+    elements = to_a
+    if default == Numeric
+      elements.my_each do |i|
+        return true unless i.class == Integer || i.class == Float || i.Complex
+      end
+      false
+    elsif default.class == Class
+      elements.my_each do |i|
+        return false if i.class == default
+      end
+      true
+    elsif default.class == Regexp
+      elements.my_each do |i|
+        return false if i =~ default
+      end
+      true
+    elsif block_given?
+      elements.my_each do |i|
+        return false if yield i
+      end
+      true
+    elsif !block_given? && !default
+      elements.my_each do |i|
+        return false if i
+      end
+      true
+    elsif default.class != Regexp && default.class != Class
+      elements.my_each do |i|
+        return false if i == default
+      end
+      true
+    end
+  end
+
+  def my_count(default = nil)
+    elements = to_a
+    count = 0
+    if default
+      elements.my_each do |i|
+        count += 1 if i == default
+      end
+    elsif block_given?
+      elements.my_each do |i|
+        count += 1 if (yield i) == true
+      end
+    else
+      elements.my_each do
+        count += 1
+      end
+    end
+    count
+  end
+
+  def my_map
+    return to_enum unless block_given?
+
+    mapped_arr = []
+    my_each { |i| mapped_arr << yield(i) }
+    mapped_arr
+  end
```

Comment:
### There are also some changes required here.

- [ ] When both a proc and a block is given, this method should only execute the proc and returns a new array with the results of running the proc for every element.

### HINT 

- A conditional statement will do the trick here.